### PR TITLE
feature/cookieless

### PR DESCRIPTION
**Pull request by osiset**
* Removes dependency on cookies
* Enables JWT as first-class
* Several internal adjustments for auth flow
* Support for XHR, jQuery, Axios, TurboLinks and others for session tokens
* Includes recent updates from `master` such as helpers moved `Utils`, CI and analysis improvements, and more

Many months of work from various contributors! PR is ready to be reviewed and merged if all is well.

### CODE REVIEW

**Comment by lucasmichot** [comment]
Man Tyler !
This is amazing ! I will 👁️ this next days 👍

File: src/ShopifyApp/Services/ChargeHelper.php [hunk 1]
```diff
@@ -267,7 +267,7 @@ public function details(Plan $plan, IShopModel $shop): PlanDetailsTransfer
         $transfer->returnUrl = URL::secure(
             Util::getShopifyConfig('billing_redirect'),
             ['plan' => $plan->getId()->toNative()]
-        );
+        )."?shop={$shop->getDomain()->toNative()}";
```
Comment (osiset) [inline src/ShopifyApp/Services/ChargeHelper.php:270]: @lucasmichot Is there a cleaner way to add a query param to URL::secure, do you know?

**Comment by osiset** [comment]
@lucasmichot I am fine with all the changes currently.

* Things seem to be in a working state
* Items can of course be refactored down the road if need-be, unless it's something urgent
* Tests are passing
* I've left one comment about query string generation

Only thing missing from this branch is the per-user abilities... I think this can be a minor release to add the feature after this big release is out.

File: src/ShopifyApp/Services/ChargeHelper.php (hunk 1 above)
Comment (lucasmichot) [reply src/ShopifyApp/Services/ChargeHelper.php:270]:
```suggestion
        ).'?'.http_build_query(['shop' => $shop->getDomain()->toNative()]);
```

Comment (lucasmichot) [reply src/ShopifyApp/Services/ChargeHelper.php:270]: I would probably use `http_build_query`

**Comment by lucasmichot** [comment]
> @lucasmichot I am fine with all the changes currently.
> 
> * Things seem to be in a working state
> * Items can of course be refactored down the road if need-be, unless it's something urgent
> * Tests are passing
> * I've left one comment about query string generation
> 
> Only thing missing from this branch is the "per-user" auth abilities... I think this can be a minor release to add the feature after this big release is out.

Hey @osiset, yes I would totally merge this and create a new major release, as lots of BC have been introduced.

Also I very agree with you on adding "per-user" auth ability later

Plenty of things can also be added/improved later - but at least the new release would have a fresh new codebase to work on 👍

**Review by lucasmichot — APPROVED** [review]
A comment added on URL generation, other than that GTM 🚀

**Comment by osiset** [comment]
> A comment added on URL generation, other than that GTM

Solid. Yeah good suggestion. Going for the merge!

**Comment by squatto** [comment]
@osiset On the [v17.0.0 release](https://github.com/osiset/laravel-shopify/releases/tag/v17.0.0) in the "Internal" section you mention `ShopContext` - should that be `SessionContext`?

**Comment by osiset** [comment]
> @osiset On the [v17.0.0 release](https://github.com/osiset/laravel-shopify/releases/tag/v17.0.0) in the "Internal" section you mention `ShopContext` - should that be `SessionContext`?

Your correct, will update :)

**Comment by squatto** [comment]
OK great, thank you! I just wanted to make sure I wasn't missing something I wasn't aware of 😬

**Pull request merged**